website/client/components/achievements/chooseClass.vue

```diff
@@ -1,119 +1,206 @@
+            .danger(@click='disableClasses(user); close();') {{ $t('optOutOfClasses') }}
 </template>
 
-<style scope>
-  .dont-despair, .death-penalty {
-    margin-top: 1.5em;
+<style lang="scss" scoped>
+  @import '~client/assets/scss/colors.scss';
+
+  .btn-primary {
+    margin-right: 1em;
+  }
+
+  .class-badge {
+    $badge-size: 32px;
+
+    width: $badge-size;
+    height: $badge-size;
+    background: $white;
+    box-shadow: 0 2px 2px 0 rgba($black, 0.16), 0 1px 4px 0 rgba($black, 0.12);
+    border-radius: 100px;
+
+    .svg-icon {
+      width: 19px;
+      height: 19px;
+    }
+  }
+
+  .class-explanation {
+    font-size: 16px;
+    margin: 1.5em auto;
+  }
+
+  .class-name {
+    font-size: 24px;
+    font-weight: bold;
+    margin: auto 0.33333em;
+  }
+
+  .danger {
+    color: $red-50;
+    margin-bottom: 0em;
+  }
+
+  .header-purple {
+    color: $purple-200;
+    margin-top: 1.33333em;
+    margin-bottom: 0em;
+  }
+
+  .modal-actions {
+    margin: 2em auto;
+  }
+
+  .selection-box {
+    width: 140px;
+    height: 148px;
+    border-radius: 16px;
+    border: solid 4px $purple-300;
+  }
+
+  .healer-color {
+    color: $yellow-10;
+  }
+
+  .rogue-color {
+    color: $purple-200;
+  }
+
+  .warrior-color {
+    color: $red-50;
+  }
+
+  .wizard-color {
+    color: $blue-10;
   }
 </style>
 
 <script>
+import axios from 'axios';
 import bModal from 'bootstrap-vue/lib/components/modal';
+import bPopover from 'bootstrap-vue/lib/components/popover';
+import changeClass from '../../../common/script/ops/changeClass';
+import disableClasses from '../../../common/script/ops/disableClasses';
 
 import Avatar from '../avatar';
 import { mapState } from 'client/libs/store';
-import percent from '../../../common/script/libs/percent';
-import {maxHealth} from '../../../common/script/index';
+import markdownDirective from 'client/directives/markdown';
+import warriorIcon from 'assets/svg/warrior.svg';
+import rogueIcon from 'assets/svg/rogue.svg';
+import healerIcon from 'assets/svg/healer.svg';
+import wizardIcon from 'assets/svg/wizard.svg';
 
 export default {
   components: {
     bModal,
+    bPopover,
     Avatar,
   },
+  computed: {
+    ...mapState({user: 'user.data'}),
+  },
   data () {
     return {
-      maxHealth,
+      heroClasses: ['warrior', 'wizard', 'rogue', 'healer'],
+      icons: Object.freeze({
+        warrior: warriorIcon,
+        rogue: rogueIcon,
+        healer: healerIcon,
+        wizard: wizardIcon,
+      }),
       selectedClass: '',
     };
   },
-  computed: {
-    ...mapState({user: 'user.data'}),
-    barStyle () {
-      return {
-        width: `${percent(this.user.stats.hp, maxHealth)}%`,
-      };
-    },
+  directives: {
+    markdown: markdownDirective,
   },
   methods: {
     close () {
       this.$root.$emit('hide::modal', 'choose-class');
     },
-    disableClasses () {
-      // @TODO:
+    async disableClasses (user) {
+      await axios.post('/api/v3/user/disable-classes');
```

Comment:
both this and changeClass should be actions on the user (store/actions/user) not defined directly in the component, also you shouldn't use await before calling the local disableClass or it'll wait for the server response and defeat the purpose of applying the changes locally so they appear immediately and not after the server response